opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -98,10 +190,57 @@ class ParallelOverlappingILU0
     //! \brief The field type of the preconditioner.
     typedef typename Domain::field_type field_type;
 
+    typedef typename matrix_type::block_type  block_type;
+    typedef typename matrix_type::size_type   size_type;
+
+protected:
+    struct CRS
+    {
+      CRS() : nRows_( 0 ) {}
+
+      size_type rows() const { return nRows_; }
+
+      size_type nonZeros() const
+      {
+        assert( rows_[ rows() ] != size_type(-1) );
+        return rows_[ rows() ];
+      }
+
+      void shrinkToFit()
+      {
+        values_.resize( nonZeros() );
+        values_.shrink_to_fit();
```

Comment:
The best way probably would be to reserve memory as you go. But my guess is that in the above one reallocation should be done.